app/graphql/types/query_type.rb

```diff
@@ -34,6 +34,11 @@ class QueryType < Types::BaseObject
       description "List of all projects"
     end
 
+    field :projects_publish_targets, [ProjectsPublishTargetType], null: true do
```

Comment:
Should this query be called "project_publish_targets" rather than "projects__publish_targets"?  And should it return a `[PublishTargetType]` rather than `[ProjectsPublishTargetType]` since there is no ProjectsPublishTarget model.

Reply:
(also see my follow-up comment in the Conversation tab)

Reply:
Remember the graphql API is not necessarily tied to the backing model - I don't think there's a need (or benefit, really) to having a model for the join entity if it has no other attributes. The `enabled` annotation here is just sugar to prevent the inspection from happening more than is necessary.

Reply:
I didn't forget.  And though a graphql API doesn't have to be tied to the backing model, we should be careful about building an API that seems to contradict the backing model.  I think this top level query will confuse developers, since I myself found it confusing!

When a Project has a PublishTarget on the backend, that means that the PublishTarget is enabled.  So why does this query, which returns a list of publish target information for the given Project, return a mixed list of enabled and disabled publish targets?

See what I'm getting at?  It all comes down to the naming here.  How about we call the type `AvailablePublishTargetType`?  I think that could help a lot!

Also, I wonder...does this even need to be a dedicated top level query?  Or could availablePublishTargets just be a field that we add to the ProjectType?

Reply:
This was really easy to implement as a single list that you replace wholesale; it would be more complicated to move part of it around - I'm reading your suggestion to be that the project type has `publish_targets` and `available_publish_targets`. Or are you thinking that available publish targets acts entirely the way it does here, as a single list? I'd be fine with that, I think.

Reply:
There is not a selectable field called `available_publish_targets` on Project that handles this.